File: solutions/two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

// Goal: Find two items in nums arr that add up to the target num

// check if nums is size 2, meaning the 2 items are the result
    // return both indexes 
// track all items in nums arr with their respective indices
// so iterate thru nums arr
    // assign each item in arr with their indices
// iterate thru nums arr again
    // get the difference between target number and item in nums
    // check if difference is found in hash, meaning it's the cooresponding number in the pair
        // return the cooresponding index from the hash and the current index

var twoSum = function(nums, target) {
    if (nums.length === 2) {
        return [0, 1];
    }
    
    let hash = {};
    
    for (let i = 0; i < nums.length; i++) {
        hash[nums[i]] = i;
    }
    
    for (let i = 0; i < nums.length; i++) {
        let difference = target - nums[i];
        if (hash[difference] && hash[difference] !== i) {            
            return [hash[difference], i];
        }
    }
        
};                                                     

File: solutions/two-sum/two-sum.py
# check if list is size 2 
#   if so then return both indices since there is only one valid solution

# track items in nums with their index with dict
# iterate thru nums list
#   assign each item with their index
# iterate thru nums list again
#   get difference between target and nums[i]
#   check to see if difference exists in dict
#       if yes then return index and dict value

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 2:
            return [0, 1]
        
        dict = {}
        
        for index, value in enumerate(nums):
            dict[value] = index
            
        for index, value in enumerate(nums):
            difference = target - value
            
            if difference in dict and dict[difference] != index:
                return [dict[difference], index]
